packages/dcos-checks/extra/dcos-checks/client/http.go

```diff
@@ -0,0 +1,36 @@
+package client
+
+import (
+	"net/http"
+
+	"github.com/dcos/dcos-go/dcos/http/transport"
+)
+
+// NewClient returns a new http client ready to handle DC/OS security.
+// A caller can optionally pass a configured *http.Client (with the right timeout for instance), in this case
+// NewClient will replace Transport.
+func NewClient(client *http.Client, iamConfig, caCert string) (*http.Client, error) {
```

Comment:
this is a very first check, we will have more and it might be needed to pass a constructed client.